src/Runner.Listener/JobDispatcher.cs

```diff
@@ -239,6 +244,13 @@ private async Task EnsureDispatchFinished(WorkerDispatcher jobDispatch, bool can
                     return;
                 }
 
+                if (this._isRunServiceJob)
+                {
+                    Trace.Error($"We are not yet checking the state of jobrequest {jobDispatch.JobId} status. Cancel running worker right away.");
+                    jobDispatch.WorkerCancellationTokenSource.Cancel();
+                    return;
+                }
```

Comment:
for run service type of job, we ignore this path, since there is no "get agent request" call, we will probably replace this with get job later on, but that's TBD